src/main/kotlin/com/amazon/opendistroforelasticsearch/indexstatemanagement/model/managedindexmetadata/ActionMetaData.kt

```diff
@@ -122,6 +133,7 @@ data class ActionMetaData(
                     FAILED -> failed = xcp.booleanValue()
                     CONSUMED_RETRIES -> consumedRetries = xcp.intValue()
                     LAST_RETRY_TIME -> lastRetryTime = if (xcp.currentToken() == Token.VALUE_NULL) null else xcp.longValue()
+                    ActionProperties.ACTION_PROPERTIES -> actionProperties = ActionProperties.parse(xcp)
```

Comment:
You're allowing ActionProperties to be null which means this will throw an exception if it does pass null into the parse. Do a similar check that's in LAST_RETRY_TIME

Reply:
Good catch, added a check.